### PR TITLE
include performance timelines with performance metrics

### PR DESCRIPTION
**Pull request by ansis**
Changes the data returned by `mapboxgl.getPerformanceMetricsAsync(...)`. This method is not public, and is only available in the debug build.

The results now include a serialized version of the performance timeline. This includes timing for each individual frame, and data on each network request made. These can be used to calculate more detailed metrics. I also removed some of the rendering metrics from gl-js. They are now calculated elsewhere to keep them in sync across versions.

The mapbox-gl-bench.js build is now minified.

There is no public impact of this pr.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] manually test the debug page
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

File: src/util/performance.js [hunk 1]
```diff
@@ -117,67 +96,25 @@ export const PerformanceUtils = {
...
-        }
-
         metrics.placementTime = placementTime;
 
         return metrics;
     },
 
     getWorkerPerformanceMetrics(): { timeOrigin: string, measures: Array<PerformanceEntry> } {
+        const entries = performance.getEntries().map(entry => {
+            const result = JSON.parse(JSON.stringify(entry));
```
Comment (mourner) [inline src/util/performance.js:106]: Nit: there's https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/toJSON

File: src/util/performance.js [hunk 2]
```diff
@@ -117,67 +96,25 @@ export const PerformanceUtils = {
...
+            if (entry.detail) {
+                Object.assign(result, {
+                    detail: entry.detail
+                });
+            }
+            return result;
+        });
         return JSON.parse(JSON.stringify({
```
Comment (mourner) [inline src/util/performance.js:114]: This won't be necessary after ^^^ too

File: src/util/worker_performance_utils.js [hunk 3]
```diff
@@ -25,7 +25,8 @@ export const WorkerPerformanceUtils = {
             const sums = {};
 
             for (const result of results) {
-                for (const measure of result.measures) {
+                const measures = result.entries.filter(e => e.entryType === 'measure');
+                for (const measure of measures) {
```
Comment (mourner) [inline src/util/worker_performance_utils.js:29]: Tiny nit — less array allocations.

```suggestion
                for (const measure of result.entries) {
                    if (measure.entryType !== 'measure') continue;
```

File: src/util/worker_performance_utils.js [hunk 4]
```diff
@@ -41,6 +42,8 @@ export const WorkerPerformanceUtils = {
 
             metrics.parseTile = metrics.parseTile1 + metrics.parseTile2;
 
+            metrics.timelines = [PerformanceUtils.getWorkerPerformanceMetrics()].concat(results);
```
Comment (mourner) [inline src/util/worker_performance_utils.js:45]: Nit — more ES-idiomatic.

```suggestion
            metrics.timelines = [PerformanceUtils.getWorkerPerformanceMetrics(), ...results];
```

**Review by mourner — APPROVED** [review]

File: src/util/performance.js (hunk 1 above)
Comment (ansis) [reply src/util/performance.js]: fixed in [17a61bb](https://github.com/mapbox/mapbox-gl-js/pull/11828/commits/17a61bbec9362d7120e3e53c1ca04ca2c511ac68)

**Review by stepankuzmin — APPROVED** [review]

**Pull request merged**